src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs

```diff
@@ -258,9 +294,51 @@ private List<string> AddArgs()
                 allArgs.Add("--logger:Console;Verbosity=" + vsTestVerbosity);
             }
 
-            if (!string.IsNullOrEmpty(this.VSTestBlame))
+            var blameCrash = !string.IsNullOrEmpty(this.VSTestBlameCrash);
+            var blameHang = !string.IsNullOrEmpty(this.VSTestBlameHang);
+            if (!string.IsNullOrEmpty(this.VSTestBlame) || blameCrash || blameHang)
             {
-                allArgs.Add("--Blame");
+                var blameArgs = "--Blame";
+
+                var dumpArgs = new List<string>();
+                if (blameCrash || blameHang)
+                {
+                    if (blameCrash)
+                    {
+                        dumpArgs.Add("CollectDump");
```

Comment:
Is it not good to initialize that type of strings as const, and use them when needed?

Reply:
Generally yes, but this is a dll that does not have many dependencies on the rest of the project and only uses the string once, so putting it in constants class does not help me not making when I re-use the string, because there is no reuse. Won't fix.